### PR TITLE
tweaks to networkd sysctl logging

### PR DESCRIPTION
**Pull request by poettering**
A followup for #32212

### CODE REVIEW

**Comment by YHNdnzj** [comment]
Hmm, if you're doing a follow-up, maybe also tackle with the comments I just left? But honestly I don't see why this is merged so early...

File: src/basic/format-ifname.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                return -EINVAL;
+
+        if (if_indextoname(ifindex, buf))
+                return 0;
+
+        if (!FLAGS_SET(flag, FORMAT_IFNAME_IFINDEX))
+                return -errno;
+
+        if (FLAGS_SET(flag, FORMAT_IFNAME_IFINDEX_WITH_PERCENT))
+                assert(snprintf_ok(buf, IF_NAMESIZE, "%%%d", ifindex));
```
Comment (YHNdnzj) [inline src/basic/format-ifname.c]: assert_se?

File: catalog/systemd.catalog.in [hunk 2]
```diff
@@ -788,18 +788,22 @@ Defined-By: systemd
...
 
-A sysctl handle under /proc/sys/net, which is managed by systemd-networkd, has been changed by another process.
-The event is raised only if the written value differs from the current one.
-The program name, the written value, the previous value, and the value initially set by networkd have been logged.
+The sysctl configuration setting @SYSCTL@, which is managed by
+systemd-networkd, has been changed by another, unrelated process
+("@OBJECT_COMM@", PID @OBJECT_PID@). This represents a conflict of ownership
+and will likely result in problems later on.
+
+Value changed to "@NEWVALUE@", value should be "@OURVALUE@".
```
Comment (YHNdnzj) [inline catalog/systemd.catalog.in]:
```suggestion
Value changed to "@NEWVALUE@", which should be "@OURVALUE@".
```

File: src/network/networkd-sysctl.c [hunk 3]
```diff
@@ -72,13 +73,13 @@ static int sysctl_event_handler(void *ctx, void *data, size_t data_sz) {
         if (!strneq(value, we->newvalue, sizeof(we->newvalue)))
                 log_struct(LOG_WARNING,
                            "MESSAGE_ID=" SD_MESSAGE_SYSCTL_CHANGED_STR,
-                           "OBJECT_PID=%d", we->pid,
+                           "OBJECT_PID=" PID_FMT, we->pid,
                            "OBJECT_COMM=%s", we->comm,
                            "SYSCTL=/proc/sys/%s", we->path,
```
Comment (YHNdnzj) [inline src/network/networkd-sysctl.c]: This can be switched to `path` btw.

File: src/resolve/resolvectl.c [hunk 4]
```diff
@@ -21,7 +21,7 @@
 #include "errno-list.h"
 #include "escape.h"
 #include "format-table.h"
-#include "format-util.h"
+#include "format-ifname.h"
```
Comment (YHNdnzj) [inline src/resolve/resolvectl.c:23]: Not sorted

File: src/basic/format-ifname.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "format-ifname.h"
+#include "string-util.h"
+
+assert_cc(STRLEN("%") + DECIMAL_STR_MAX(int) <= IF_NAMESIZE);
```
Comment (YHNdnzj) [inline src/basic/format-ifname.c:6]: I'd insert a newline below, but doesn't matter

**Review by YHNdnzj — APPROVED** [review]

**Review by teknoraver — APPROVED** [review]

**Pull request merged**